internal/util/search/search.go

```diff
@@ -83,26 +90,46 @@ func (sr *SearchReplace) Perform(resourcesPath string) error {
 
 	return kio.Pipeline{
 		Inputs:  []kio.Reader{inout},
-		Filters: []kio.Filter{kio.FilterAll(sr)},
+		Filters: []kio.Filter{filterAll(sr)},
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 }
 
 // Filter parses input node and performs search and replace operation on the node
 func (sr *SearchReplace) Filter(object *yaml.RNode) (*yaml.RNode, error) {
+	originalObject, err := deepCopy(object)
+	if err != nil {
+		return object, err
+	}
 	filePath, _, err := kioutil.GetFileAnnotations(object)
 	if err != nil {
 		return object, err
 	}
 	sr.filePath = filePath
 
 	if sr.shouldPutLiteralByPath() {
-		return object, sr.putLiteral(object)
+		err = sr.putLiteral(object)
+	} else {
+		// traverse the node to perform search/put operation
+		err = accept(sr, object)
+	}
+
+	if err != nil {
+		return object, err
+	}
+
+	if sr.Validate {
+		validateResult, err := ValidateNode(object)
```

Comment:
So this seems to separate the validation of fields from the process of determining the correct type. And the tests seems to all focus on validation errors involving incorrect types. Should we do validation at the same time as we determine the type or are there validation errors we can't catch at that point?

This also seems to do validation on the complete package. Do we want to do that? It seems like maybe this should work for users even if there might be other fields in any of the resources that doesn't validate.